FHEM/14_LTECH.pm

```diff
@@ -0,0 +1,516 @@
+        }
+        return
+	}
+    elsif($cmd eq "off") {
+			Send ($hash, $name, "FE00000000FF00" );
+            readingsSingleUpdate($hash, 'rgbcolor', '000000', 1);
+            return
+	}
+    elsif($cmd eq "h") {
+        if( looks_like_number($value) && 0 <= $value  && $value <= 360 ){
+            my $rgbcolor = ReadingsVal( $name, 'rgbcolor', 'FF3700' );
+            my @rgb = Color::hex2rgb($rgbcolor);
+            my ($h ,$s ,$v ) = Color::rgb2hsv( $rgb[0] / 255, $rgb[1] / 255, $rgb[2] / 255 );
+            $h = $value / 360;
+            my $rgbcolor_sel = Color::hsv2hex($h,$s,$v);
+            readingsSingleUpdate($hash, 'rgbcolor_sel', $rgbcolor_sel, 1);
+            $args[0] = "on"; 
+            Set($hash, $name, @args);
+            return
+        }
+        return "Please provide valid number between 0 and 360"  
+	}
+    elsif($cmd eq "brightness") {
+        if( looks_like_number($value) && 0 <= $value  && $value <= 100 ){
+            my $rgbcolor = ReadingsVal( $name, 'rgbcolor', 'FF3700' );
+            my @rgb = Color::hex2rgb($rgbcolor);
+            my ($h ,$s ,$v ) = Color::rgb2hsv( $rgb[0] / 255, $rgb[1] / 255, $rgb[2] / 255 );
+            $v = $value / 100;
+            my $rgbcolor_sel = Color::hsv2hex($h,$s,$v);
+            readingsSingleUpdate($hash, 'rgbcolor_sel', $rgbcolor_sel, 1);
+            $args[0] = "on"; 
+            Set($hash, $name, @args);
+            return
+        }
+        return "Please provide valid number between 0 and 100"
+    }
+    elsif($cmd eq "saturation") {
+        if( looks_like_number($value) && 0 <= $value  && $value <= 100 ){
+            my $rgbcolor = ReadingsVal( $name, 'rgbcolor', 'FF3700' );
+            my @rgb = Color::hex2rgb($rgbcolor);
+            my ($h ,$s ,$v ) = Color::rgb2hsv( $rgb[0] / 255, $rgb[1] / 255, $rgb[2] / 255 );
+            $s = $value / 100;
+            my $rgbcolor_sel = Color::hsv2hex($h,$s,$v);
+            readingsSingleUpdate($hash, 'rgbcolor_sel', $rgbcolor_sel, 1);
+            $args[0] = "on"; 
+            Set($hash, $name, @args);
+            return
+        }
+        return "Please provide valid number between 0 and 100"
+    }
+
+
+	my @cList;
+	my $atts = AttrVal( $name, 'setList', "" );
+	my %setlist = split( "[: ][ ]*", $atts );
+	foreach my $k ( sort keys %sets ) {
+		my $opts = undef;
+		$opts = $sets{$k};
+		$opts = $setlist{$k} if ( exists( $setlist{$k} ) );
+		if ( defined($opts) ) {
+			push( @cList, $k . ':' . $opts );
+		}
+		else {
+			push( @cList, $k );
+		}
+	}    # end foreach
+	return "Unknown argument $cmd, choose one of " . join( " ", @cList );
+}
+#############################
+sub Icon {
+   my ($name,$icon) = @_;
+   my $hash = $defs{$name};
+   my $state = ReadingsVal( $name, 'brightness', 'undef' );
+   my $rgbcolor = ReadingsVal( $name, 'rgbcolor', 'undef' );
+   $hash->{STATE} = $state / 2.55 ;
+   my $sticon = ".*:light_light_dim_". ($state / 2.55)."0\@".$rgbcolor;
+   return $sticon ;
```

Comment:
wenn in {DEFINED} der Wert "defined" steht, gibt das bestimmt einen Fehler.

Ist das Berechnen eines Icons aber nicht eher etwas für den Anwender über z.B. StateFormat usw. ?

Reply:
Obsolete